forge-ai/src/main/java/forge/ai/AiAttackController.java

```diff
@@ -795,14 +796,14 @@ public final int declareAttackers(final Combat combat) {
                         && isEffectiveAttacker(ai, attacker, combat, defender)) {
                     mustAttack = true;
                 } else if (seasonOfTheWitch) {
-                    // TODO: if there are other ways to tap this creature (like mana creature), then don't need to attack
+                    //TODO: if there are other ways to tap this creature (like mana creature), then don't need to attack
                     mustAttack = true;
                 } else {
-                    // TODO move to static Ability
-                    if (attacker.hasKeyword("CARDNAME attacks each combat if able.") || attacker.hasStartOfKeyword("CARDNAME attacks specific player each combat if able")) {
-                        // TODO switch defender if there's one without a cost or it's not the specific player
```

Comment:
can you leave this TODO in for @Agetian or me later?

The problem is the AttackRequirement might not count if opponent has Propaganda-stuff

but in multiplayer there might be another defender without that invalidates that instead

so minor potential for AI to cheat out of it

Rest looks pretty solid!